infra/modules/database_layer/db.tf

```
@@ -18,6 +18,7 @@ resource "aws_db_subnet_group" "default" {
 }
 
 resource "aws_db_instance" "db" {
+  count      = "${(var.db_instance_enabled ? 1 : 0)}"
```

```suggestion
  count      = "${var.db_instance_enabled ? 1 : 0}"
```

Style-wise, enclosing brackets aren't used in the rest of the code for simple enable / disable count expressions.